utils/common/src/main/java/org/apache/brooklyn/util/time/DurationPredicates.java

```diff
@@ -0,0 +1,119 @@
+     *         to {@link Predicate#apply(Object)} is positive.
+     */
+    public static Predicate<Duration> isPositive() {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration duration) {
+                return duration != null && duration.isPositive();
+            }
+        };
+    }
+
+    /**
+     * @return A {@link Predicate} that checks if the {@link Duration} supplied
+     *         to {@link Predicate#apply(Object)} is negative.
+     */
+    public static Predicate<Duration> isNegative() {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration duration) {
+                return duration != null && duration.isNegative();
+            }
+        };
+    }
+
+    /**
+     * @param duration
+     *            The {@link Duration} that will be the basis for comparison in
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is longer than any {@link Duration}
+     *         supplied to to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Duration> isLongerThanDuration(final Duration duration) {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration other) {
+                return duration != null && other != null && duration.isLongerThan(other);
+            }
+        };
+    }
+
+    /**
+     * @param duration
+     *            The {@link Duration} that will be the basis for comparison in
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is longer than any {@link Stopwatch}
+     *         supplied to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Stopwatch> isLongerThanStopwatch(final Duration duration) {
+        return new Predicate<Stopwatch>() {
+            @Override
+            public boolean apply(Stopwatch other) {
+                return duration != null && other != null && duration.isLongerThan(other);
+            }
+        };
+    }
+
+    /**
+     * @param duration
+     *            The {@link Duration} that will be the basis for comparison in
+     *            the returned {@link Predicate}.
+     * @return A {@link Predicate} that checks if the {@link Duration} that was
+     *         supplied to this method is shorter than any {@link Duration}
+     *         supplied to to the returned {@link Predicate#apply(Object)}.
+     */
+    public static Predicate<Duration> isShorterThanDuration(final Duration duration) {
+        return new Predicate<Duration>() {
+            @Override
+            public boolean apply(Duration other) {
+                return duration != null && other != null && duration.isShorterThan(other);
```

Comment:
Again, the `Duration` the predicate speaks about here is the one called `other`, the `duration` is constant over the life of the predicate.